src/EventListener/SessionListener.php

```diff
@@ -112,4 +116,31 @@ public static function getSubscribedEvents()
     {
         return AbstractSessionListener::getSubscribedEvents();
     }
+
+    /**
+     * Mark the response as uncachable if it has a non-session cookie.
+     *
+     * @param Response $response
+     */
+    private function handleResponseCookies(Response $response)
+    {
+        /** @var Cookie $cookie */
+        foreach ($response->headers->getCookies() as $cookie) {
+            // Move session cookie from Symfony response to PHP headers
+            if (session_name() === $cookie->getName()) {
+                $response->headers->removeCookie($cookie->getName(), $cookie->getPath(), $cookie->getDomain());
+                \header((string) $cookie);
+                continue;
+            }
+
+            if ($response->isCacheable()) {
+                $response
+                    ->setPrivate()
+                    ->setMaxAge(0)
+                    ->headers->addCacheControlDirective('must-revalidate');
+            }
+
+            break;
```

Comment:
If the session cookie is the third cookie it would not get removed because of this line. Is that on purpose?

Reply:
You're absolutely right, fixed in add696b0246abf2d8134747d67f8ead3e197a9bd